src/util/file-downloader.ts

```diff
@@ -0,0 +1,66 @@
+import { IHttpClient } from '../sdk/http';
+import { IDownloadFileProvider } from '../sdk/resources';
+
+const path = require('path');
+const fs = require('fs');
+
+export class FileDownloader {
+	constructor(private httpClient: IHttpClient, private fileProvider: IDownloadFileProvider) {
+	}
+
+	download(id: string, outputDirectory: string): Promise<string> {
+		return new Promise<any>(async (resolve, reject) => {
+			try {
+				let _this = this;
+				this.createDirectory(outputDirectory);
+				let requestPath = this.fileProvider.getDownloadRequestPath(id);
+
+				return await this.httpClient.download(requestPath, async (response) => {
+					if (response.statusCode === 404) {
```

Comment:
This logic should be handled in the http client. We do not want the consumer of `download` to be aware of the implementation details of the response.